src/common/commonutils/FileUtils.c

```diff
@@ -1331,6 +1331,7 @@ int CheckTextNotFoundInEnvironmentVariable(const char* variableName, const char*
                         OsConfigCaptureReason(reason, "'%s' found in '%s' ('%s')", text, variableName, found);
                         foundText = true;
                         status = EEXIST;
+                        break;
```

Comment:
>break; [](http://example.com/codeflow?start=24&length=6)

The existing code's intention is to continue to search all instances in the text. Here it record a match was found but does not break, continues. That's in order for all instances to be logged and captured in the reason (as opposed to bail out at the first found instance). Does this make sense? Please let me know if you need to discuss this further.

Reply:
Do we need this logging? Continuing iteration is waste of resources if there is no specific requirement.

Reply:
To follow up on my last reply (can you please answer that question? thx) - if it's just a transient delay in fuzzing, that is fine, is due to how this function works. However, if it ends into a denial of service problem with an infinite loop, then we need a fix to harden the function or we need to skip this function from fuzzing.

The hardening fix would be not to change the internal algorithm of the function, but instead before executing that, check the file size and if is bigger than a certain size, bail out right there, reporting as reason that the file is too large to be processed.